### PR TITLE
745 add examples

### LINKED ISSUE #767: Clarify purpose of `format_reason_default`
Tester feedback:
> Not sure where to use this format as DCTREASP is created from derive_disposition_reason. Hence not checked this function

It should be clarified that it is not intended that the function is called directly but that it is used as input to `derive_disposition_reason()`.

A "See also" section should be added, too.

### PR DESCRIPTION
**Pull request by tomratford**
closes #767

in line with #767 the examples for `format_reason_default` and `format_eoxxstt_default` is their parent function call to avoid confusion.

dev utilities `extend_source_datasets` and `filter_date_sources` were the only other functions without examples so I have added these based off other examples used within the other tte function examples.

---

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://roche-gsk.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://roche-gsk.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://roche-gsk.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::documents()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://roche-gsk.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: R/derive_param_tte.R [hunk 1]
```diff
@@ -469,6 +469,45 @@ derive_param_tte <- function(dataset = NULL,
...
 #' @keywords dev_utility
 #'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(lubridate, warn.conflicts = FALSE)
+#'
+#' adsl <- tibble::tribble(
+#' ~USUBJID, ~TRTSDT,           ~EOSDT,
+#' "01",     ymd("2020-12-06"), ymd("2021-03-06"),
+#' "02",     ymd("2021-01-16"), ymd("2021-02-03")
```
Comment (thomas-neitmann) [inline R/derive_param_tte.R:479]: Please indent by 2 spaces.

File: R/derive_param_tte.R [hunk 2]
```diff
@@ -576,6 +615,30 @@ filter_date_sources <- function(sources,
...
```
Comment (thomas-neitmann) [inline R/derive_param_tte.R:625]: Please indent by 2 spaces.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

**Review by bms63 — APPROVED** [review]
@thomas-neitmann I think this looks good - can we squeeze into 0.6.0 release?

**Comment by bms63** [comment]
@tomratford We might try and sneak this into the 0.6.0 release.  CAn you update the news please if they has not been done?

File: NEWS.md [hunk 3]
```diff
@@ -64,6 +64,8 @@ this case the day is imputed as `15`. (#592).
 
 - Updated [Pull Request Review Guidance](../articles/pr_review_guidance.html) on using `task-list-completed` workflow. (#817)
 
+- Examples have been added for `format_reason_default`, `format_eoxxstt_default`, `extend_source_datasets` and `filter_date_sources` (#745)
```
Comment (thomas-neitmann) [inline NEWS.md:67]: Please add a set of parentheses, i.e. `()`, after the function names.

File: R/derive_param_tte.R [hunk 4]
```diff
@@ -576,6 +615,30 @@ filter_date_sources <- function(sources,
 #'
 #' @keywords dev_utility
 #'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(lubridate, warn.conflicts = FALSE)
```
Comment (thomas-neitmann) [inline R/derive_param_tte.R:620]: Same as above.

File: R/derive_var_disposition_status.R [hunk 5]
```diff
@@ -139,6 +140,22 @@ derive_disposition_status <- function(dataset,
...
 #' @keywords user_utility adsl computation
+#' @seealso [derive_disposition_status()]
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(admiral.test)
+#' data("dm")
+#' data("ds")
+#'
+#' dm %>%
+#'   derive_disposition_status(
```
Comment (thomas-neitmann) [inline R/derive_var_disposition_status.R:151]: See comment above.

File: R/derive_vars_disposition_reason.R [hunk 6]
```diff
@@ -162,7 +162,8 @@ derive_disposition_reason <- function(dataset,
 
 #' Default format for the disposition reason
 #'
-#' Define a function to map the disposition reason
+#' Define a function to map the disposition reason, to be used as a parameter in
+#' `derive_disposition_reason()`.
```
Comment (thomas-neitmann) [inline R/derive_vars_disposition_reason.R:166]: The function has ben renamed to `derive_vars_disposition_reason()`.

File: R/derive_vars_disposition_reason.R [hunk 7]
```diff
@@ -173,17 +174,29 @@ derive_disposition_reason <- function(dataset,
...
-#' DCSREAS = format_reason_default(DSDECOD)
-#' DCSREASP = format_reason_default(DSDECOD, DSTERM)
-#' ```
-#'
 #' @return A `character` vector
 #'
 #' @author Samia Kabi
 #' @export
 #' @keywords user_utility adsl computation
+#' @seealso [derive_disposition_reason()]
```
Comment (thomas-neitmann) [inline R/derive_vars_disposition_reason.R:182]: See comment above.

File: R/derive_var_disposition_status.R [hunk 8]
```diff
@@ -139,6 +140,22 @@ derive_disposition_status <- function(dataset,
 #' @author Samia Kabi
 #' @export
 #' @keywords user_utility adsl computation
+#' @seealso [derive_disposition_status()]
```
Comment (thomas-neitmann) [inline R/derive_var_disposition_status.R:143]: See comment above.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

**Review by thomas-neitmann — APPROVED** [review]
Thank you!

**Pull request merged**